api/providers/views.py

```diff
@@ -688,13 +689,20 @@ def get_default_queryset(self):
         default_schema_id = provider.default_schema.id if provider.default_schema else None
         if not default_schema_id:
             return provider.schemas.get_latest_versions(request=self.request, invisible=True).filter(active=True)
-        schemas = provider.schemas.values('name').annotate(version=Max('schema_version'))
-        annotated_schemas = schemas.annotate(default_schema_ordering=Case(
+        latest_versions = RegistrationSchema.objects.values('name').annotate(latest_version=Max('schema_version'))
+        annotated = RegistrationSchema.objects.all().annotate(
+            latest_version=Subquery(
+                latest_versions.filter(name=OuterRef('name')).values('latest_version')[:1],
+                output_field=IntegerField(),
+            ),
+        )
+        filtered = annotated.filter(schema_version=F('latest_version'), providers=provider)
```

Comment:
I would update the existing query for `get_latest_version` to use this approach? then this function can just do the optional annotation to surface the default. you can also feel free to compress some of this logic

Reply:
just a reminder that this can be removed now that `provider.schemas.get_latest_versions` implements this logic